src/color.rs

```diff
@@ -1187,32 +1277,60 @@ fn parse_rgb<'i, 't, P>(
-        NumberOrPercentage::Number { value } => (clamp_floor_256_f32(value), true),
-        NumberOrPercentage::Percentage { unit_value } => (clamp_unit_f32(unit_value), false),
-    };
-
-    let uses_commas = arguments.try_parse(|i| i.expect_comma()).is_ok();
-
-    let green;
-    let blue;
-    if is_number {
-        green = clamp_floor_256_f32(color_parser.parse_number(arguments)?);
-        if uses_commas {
-            arguments.expect_comma()?;
+    let maybe_red = parse_none_or(arguments, |p| color_parser.parse_number_or_percentage(p))?;
+
+    // If the first component is not "none" and is followed by a comma, then we
+    // are parsing the legacy syntax.
+    let is_legacy_syntax = maybe_red.is_some() && arguments.try_parse(|p| p.expect_comma()).is_ok();
+
+    let red: u8;
+    let green: u8;
+    let blue: u8;
+
+    let alpha = if is_legacy_syntax {
+        match maybe_red.unwrap() {
+            NumberOrPercentage::Number { value } => {
+                red = clamp_floor_256_f32(value);
+                green = clamp_floor_256_f32(color_parser.parse_number(arguments)?);
+                arguments.expect_comma()?;
+                blue = clamp_floor_256_f32(color_parser.parse_number(arguments)?);
+            }
+            NumberOrPercentage::Percentage { unit_value } => {
+                red = clamp_unit_f32(unit_value);
+                green = clamp_unit_f32(color_parser.parse_percentage(arguments)?);
+                arguments.expect_comma()?;
+                blue = clamp_unit_f32(color_parser.parse_percentage(arguments)?);
+            }
         }
-        blue = clamp_floor_256_f32(color_parser.parse_number(arguments)?);
+
+        parse_legacy_alpha(color_parser, arguments)?
     } else {
-        green = clamp_unit_f32(color_parser.parse_percentage(arguments)?);
-        if uses_commas {
-            arguments.expect_comma()?;
+        #[inline]
+        fn get_component_value(c: &Option<NumberOrPercentage>) -> u8 {
+            match *c {
+                Some(NumberOrPercentage::Number { value }) => clamp_floor_256_f32(value),
+                Some(NumberOrPercentage::Percentage { unit_value }) => clamp_unit_f32(unit_value),
+                None => 0,
+            }
         }
-        blue = clamp_unit_f32(color_parser.parse_percentage(arguments)?);
-    }
 
-    let alpha = parse_legacy_alpha(color_parser, arguments, uses_commas)?;
+        red = get_component_value(&maybe_red);
+
+        green = get_component_value(&parse_none_or(arguments, |p| {
+            color_parser.parse_number_or_percentage(p)
+        })?);
+
+        blue = get_component_value(&parse_none_or(arguments, |p| {
+            color_parser.parse_number_or_percentage(p)
+        })?);
+
+        if !arguments.is_exhausted() {
+            arguments.expect_delim('/')?;
+            parse_none_or(arguments, |p| parse_alpha_component(color_parser, p))?.unwrap_or(0.0)
```

Comment:
I'm confused, why don't we need to preserve `None` in `rgba`? Seems to me we should (even if it should serialize when resolved as zero).

Reply:
Its useless information right now.  Perhaps when relative color syntax is added it is needed, but I haven't looked so far into that.  Any particular reason you think it should be added even if its not used?

Reply:
You need it for interpolation don't you?

`color-mix(in srgb, rgba(100 100 100 / 1.0) 50%, rgba(0 none 0 / 1.0))` should be `rgba(50, 100, 50, 1.0)` not `rgba(50, 50, 50, 1.0)`, right?

Reply:
Added `none` support to rgb and hsl as well.